### ISSUE TITLE
Download specific UCR_UEA dataset

### ISSUE THREAD

**Issue by rth** [rtavenar/tslearn#22]
It would be nice if it was possible to download a specific (e.g. small) dataset when running,
```
d = UCR_UEA_datasets()
X_train, y_train, X_test, y_test = d.load_dataset('<some-id>')
```
as opposed to downloading everything and caching it. It particular, that would make it possible to use this in some gallery example as well.

**Edit:** moved part of this comment to a separate issue https://github.com/rtavenar/tslearn/issues/23 as requested.

**Comment by rtavenar** [comment]
Hi, and thanks for your feedback!

Could you split this in two separate issues, since I think there are two things discussed here.

Let's keep this issue for the first problem you point out: downloading separate datasets.
At the moment, what the code does is download an archive from www.timeseriesclassification.com, unzip it locally and access datasets locally.
In order to do what you suggest, we should host the datasets by ourselves (we should ask UCR/UEA for permission, I guess + we should decide on where to host the data, and host to pay for the hosting service :) ).

Maybe that could be a good solution (if UCR/UEA guys are OK, once again), but I have no idea how large a single dataset in .npz format could be.

**Comment by rth** [comment]
Thanks for your response @rtavenar !

> Could you split this in two separate issues, since I think there are two things discussed here.

Done, moved the second part to a separate issue https://github.com/rtavenar/tslearn/issues/23

>  In order to do what you suggest, we should host the datasets by ourselves (we should ask UCR/UEA for permission, I guess + we should decide on where to host the data, and how to pay for the hosting service :) ).

Well generally it's a bit of a pain to host datasets. I know that scikit-learn moved a copy of datasets it depends on at https://figshare.com/ (https://github.com/scikit-learn/scikit-learn/issues/7425)

But in this issue I was just thinking that as far as I saw www.timeseriesclassification.com does provide a fairly straightforward download link for each dataset, e.g.

http://www.timeseriesclassification.com/Downloads/DiatomSizeReduction.zip

which is 
```py
"http://www.timeseriesclassification.com/Downloads/%s.zip" % dataset_name
```

so I guess it's possible to only download those datasets that are explicitly loaded (then cache them as it is done currently).  I'have not checked whether this generalizes to all datasets though...

**Comment by rtavenar** [comment]
Oh, great! I didn't know that.

Then we should have a way to deal with it, definitely.

**Comment by rtavenar** [comment]
This behaviour is now implemented in `tslearn` (>= 0.1.12)
